### ISSUE TITLE
youtube_dl error on playback

### ISSUE THREAD

**Issue by CrowsOfWar** [mps-youtube/mps-youtube#598]
Crash:
```
Traceback (most recent call last):
  File "E:\Devel\src\git\mps-youtube\mps_youtube\player.py", line 148, in _playsong
  File "E:\Devel\src\git\mps-youtube\mps_youtube\streams.py", line 51, in get
  File "E:\Devel\src\git\mps-youtube\mps_youtube\util.py", line 201, in get_pafy
  File "E:\Devel\src\git\mps-youtube\pafy\pafy.py", line 122, in new
  File "E:\Devel\src\git\mps-youtube\pafy\backend_youtube_dl.py", line 29, in __init__
  File "E:\Devel\src\git\mps-youtube\pafy\backend_shared.py", line 95, in __init__
  File "E:\Devel\src\git\mps-youtube\pafy\backend_youtube_dl.py", line 41, in _fetch_basic
OSError: ERROR: Signature extraction failed: Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\youtube_dl\extractor\youtube.py", line 1001, in _decrypt_signature
  File "C:\Python34\lib\site-packages\youtube_dl\extractor\youtube.py", line 915, in _extract_signature_function
  File "C:\Python34\lib\site-packages\youtube_dl\extractor\common.py", line 503, in _download_webpage
  File "C:\Python34\lib\site-packages\youtube_dl\extractor\common.py", line 410, in _download_webpage_handle
  File "C:\Python34\lib\site-packages\youtube_dl\extractor\common.py", line 390, in _request_webpage
  File "C:\Python34\lib\site-packages\youtube_dl\YoutubeDL.py", line 1954, in urlopen
  File "C:\Python34\lib\site-packages\youtube_dl\utils.py", line 499, in sanitized_Request
  File "C:\Python34\lib\urllib\request.py", line 266, in __init__
  File "C:\Python34\lib\urllib\request.py", line 292, in full_url
  File "C:\Python34\lib\urllib\request.py", line 321, in _parse
ValueError: unknown url type: '/yts/jsbin/player-en_US-vflDQGgxm/base.js'
 (caused by ValueError("unknown url type: '/yts/jsbin/player-en_US-vflDQGgxm/base.js'",));
```

Installed the latest EXE release on Windows. I set up mpv in the same folder and set the player succesfully. After searching for songs via '/' and pressing the correct number, this error displays.

**Comment by LNFWebsite** [comment]
Same error on my end...

It's an infinite loop that I believe is caused by mps-youtube using an old version of youtube-dl.

https://github.com/rg3/youtube-dl/issues/4232

^ I'll throw this here for the developer ^

**Comment by ceebu** [comment]
The link to [youtube-dl error](https://github.com/rg3/youtube-dl/issues/4232) shows using an old version of youtube-dl as the cause of the error. ON other similar errors [11663](https://github.com/rg3/youtube-dl/issues/11663), looks like it was fixed by 10th Jan.

1. I updated youtube-dl to the latest verstion by running "youtube-dl.exe -U".  Also, updated MPV to the latest version. After being updated to 2017.03.10, i still got this error when trying to play a song. 
2. Using the same URL on YTDL downloaded the file to my PC. (this confirmed YTDL was working)
3. Using MPSYT to download the song, downloaded the file. (So playing was the problem, not downlaoding)
4. Playing the song again through MPSYT started working !! (even though I got an error once without interrupting playback)

In a nutshell, did upgrading YTDL / MPV fix this ? may-be.. YMMV

**Comment by LNFWebsite** [comment]
@ceebu @ids1024 I've manually updated the `youtube-dl.exe` executable that is placed in the same directory as MPSYT by downloading the latest, but it seems as if MPSYT packages youtube-dl within it's own executable (`mpsyt-0.2.7.1.exe`)... Either that or I'm all wet.

I first extracted mpv.exe into the MPSYT directory, then I downloaded the youtube-dl.exe executable into the same directory (per instructions for mpv.exe installation). Same error.

So the only thing I can figure is that MPSYT uses an old version within it's own code, and is not tied to MPV's usage of the youtube-dl file.

I might try the docker container installation at some point. The original dev doesn't seem very focused on issues at the moment...

**Comment by ids1024** [comment]
> So the only thing I can figure is that MPSYT uses an old version within it's own code, and is not tied to MPV's usage of the youtube-dl file.

Yes, that is indeed the case. I've been meaning to address this, but that requires booting Windows to build a new binary.

**Comment by ids1024** [comment]
You can also install mps-youtube with pip: `pip3 install mps-youtube youtube-dl`

**Comment by ids1024** [comment]
I've updated the `.exe` attached to the latest release. Hopefully that will fix this.

**Comment by LNFWebsite** [comment]
@ids1024 Thank you! I'll be downloading it right now to see how it's working!

**Comment by CrowsOfWar** [comment]
@ids1024 Thanks! I'm kinda stupid and can't find the latest release, do you have a link?

**Comment by LNFWebsite** [comment]
@CrowsOfWar Sure... It's right under releases under the latest release....

Here's a direct link: https://github.com/mps-youtube/mps-youtube/releases/download/v0.2.7.1/mpsyt-0.2.7.1.exe

**Comment by CrowsOfWar** [comment]
Thanks.
